automated-demo/auto-demo.sh

```diff
@@ -431,39 +429,46 @@ mongo_pacman_demo_cleanup()
-  run_ok_or_fail "oc --context=feddemocl2 -n ${DEMO_NAMESPACE} delete route mongo" "0" "1" "1"
-  run_ok_or_fail "oc --context=feddemocl3 -n ${DEMO_NAMESPACE} delete route mongo" "0" "1" "1"
+  run_ok_or_fail "oc --context=feddemocl1 delete federatednamespace ${DEMO_NAMESPACE} -n ${DEMO_NAMESPACE}" "0" "1" "1"
   echo "Deleting Federation"
   echo "Disabling federated resources on Host Cluster (may take a while)"
-  for type in namespaces ingresses.extensions secrets serviceaccounts services configmaps persistentvolumeclaims deployments.apps
+  for type in namespaces ingresses.extensions secrets serviceaccounts services configmaps persistentvolumeclaims deployments.apps roles.rbac.authorization.k8s.io rolebindings.rbac.authorization.k8s.io clusterrolebindings.rbac.authorization.k8s.io clusterroles.rbac.authorization.k8s.io
   do
-    run_ok_or_fail "./bin/kubefedctl disable "${type}" --kubefed-namespace=${DEMO_NAMESPACE} --host-cluster-context=feddemocl1" "0" "1" "1"
+    run_ok_or_fail "./bin/kubefedctl disable "${type}" --host-cluster-context feddemocl1" "0" "1" "1"
   done
   echo "Deleting Subscription"
-  run_ok_or_fail "oc --context=feddemocl1 -n ${DEMO_NAMESPACE} delete subscription federation" "0" "1"
+  run_ok_or_fail "oc --context=feddemocl1 -n kube-federation-system delete subscription federation" "0" "1"
   echo "Deleting ClusterServiceVersion"
-  run_ok_or_fail "oc --context=feddemocl1 -n ${DEMO_NAMESPACE} delete csv kubefed-operator.${CSV_RELEASE}" "0" "1"
+  run_ok_or_fail "oc --context=feddemocl1 -n kube-federation-system delete csv kubefed-operator.${CSV_RELEASE}" "0" "1"
   echo "Deleting CatalogSourceConfig"
-  run_ok_or_fail "oc --context=feddemocl1 -n openshift-marketplace delete catalogsourceconfig installed-federation-${DEMO_NAMESPACE}" "0" "1"
+  run_ok_or_fail "oc --context=feddemocl1 -n openshift-marketplace delete catalogsourceconfig installed-federation-kube-federation-system" "0" "1"
   echo "Deleting OperatorGroup"
-  run_ok_or_fail "oc --context=feddemocl1 -n ${DEMO_NAMESPACE} delete operatorgroup federation" "0" "1"
+  run_ok_or_fail "oc --context=feddemocl1 -n kube-federation-system delete operatorgroup federation" "0" "1"
 }
 
 namespace_kubefed_cleanup()
 {
-  for cluster in feddemocl1 feddemocl2 feddemocl3
-  do
-    echo "Removing namespace ${namespace} from cluster ${cluster}"
-    run_ok_or_fail "oc --context=${cluster} delete namespace ${DEMO_NAMESPACE}" "0" "1"
-  done 
+  echo "Removing namespace from cluster"
+  run_ok_or_fail "oc --context=feddemocl1 delete namespace kube-federation-system" "0" "1"
+  run_ok_or_fail "oc --context=feddemocl2 delete namespace kube-federation-system" "0" "1"
+  run_ok_or_fail "oc --context=feddemocl3 delete namespace kube-federation-system" "0" "1"
+  run_ok_or_fail "oc --context=feddemocl1 delete namespace ${DEMO_NAMESPACE}" "0" "1"
+  run_ok_or_fail "oc --context=feddemocl2 delete namespace ${DEMO_NAMESPACE}" "0" "1"
```

Comment:
I believe propagated namespaced are deleted when the associated federatednamespace is deleted.